internal/handlers/handler_firstfactor.go

```diff
@@ -127,12 +127,16 @@ func FirstFactorPost(msInitialDelay time.Duration, delayEnabled bool) middleware
 
 		ctx.Logger.Debugf("Credentials validation of user %s is ok", bodyJSON.Username)
 
-		// Reset all values from previous session before regenerating the cookie.
-		err = ctx.SaveSession(session.NewDefaultUserSession())
+		userSession := ctx.GetSession()
 
-		if err != nil {
-			handleAuthenticationUnauthorized(ctx, fmt.Errorf("Unable to reset the session for user %s: %s", bodyJSON.Username, err.Error()), authenticationFailedMessage)
-			return
+		if userSession.OIDCWorkflowSession == nil {
+			// Reset all values from previous session before regenerating the cookie.
+			err = ctx.SaveSession(session.NewDefaultUserSession())
```

Comment:
Why not also do it if worklow is not nil? I might have written this myself but It's not clear to me anymore reading the code. I'll need to figure out why I did that and add a comment or just remove that condition.

Reply:
Basically if we do that here when the workflow is not nil we lose the workflow completely, and since it's connected to the session not the user, we can reasonably assume the browser they're using is the same as the one that started the workflow.

The only other alternative here is to store the OIDC workflow value and reapply it, otherwise the user can never actually consent to OIDC.

Reply:
I added some comments to document this. See what you think.

Reply:
I will do a proposal to avoid the if but keep the OIDC workflow.